tokio/src/sync/broadcast.rs

```diff
@@ -579,17 +579,24 @@ impl<T> Sender<T> {
         let slot = &self.shared.buffer[idx];
 
         // Acquire the write lock
-        let mut prev = slot.lock.fetch_add(1, SeqCst);
+        let mut prev = slot.lock.fetch_or(1, SeqCst);
 
         while prev & !1 != 0 {
             // Concurrent readers, we must go to sleep
             tail = self.shared.condvar.wait(tail).unwrap();
 
             prev = slot.lock.load(SeqCst);
+
+            if prev & 1 == 0 {
+                // A newer write completed in the meantime.
+                return Ok(rem);
```

Comment:
This return here isn't entirely clear to me. Could you elaborate on why returning `Ok` without sending the value is OK here?

Reply:
I think I see what this line is doing, but it would be helpful to add a comment explaining the failure scenario and why returning`Ok` is acceptable.

Reply:
The value has already been overwritten by another thread calling send on the same slot. So there is nothing left to write into the slot.

Reply:
Right, I realized this after the fact. It would be :100: to add a comment explaining this as it is not obvious :)

Reply:
Done.